src/app/containers/teachers-container/teachers/teacher-card/teacher-card.component.ts

```diff
@@ -0,0 +1,86 @@
+import { NotificationService } from '../../../../services/notification.service';
+import { TeachersService } from '../../../../services/teachers.service';
+import { Teacher } from '../../../../models/teacher.model';
+import { Component, OnInit, Input } from '@angular/core';
+import { Validators, FormBuilder, AbstractControl } from '@angular/forms';
+import { MAT_MOMENT_DATE_FORMATS, MomentDateAdapter } from '@angular/material-moment-adapter';
+import { DateAdapter, MAT_DATE_FORMATS, MAT_DATE_LOCALE } from '@angular/material/core';
+import { ValidationService } from '../../../../services/validation.service';
+@Component({
+  selector: 'webui-teacher-card',
+  templateUrl: './teacher-card.component.html',
+  styleUrls: ['./teacher-card.component.scss'],
+  providers: [
+    {provide: MAT_DATE_LOCALE, useValue: 'uk-UK'},
+    {provide: DateAdapter, useClass: MomentDateAdapter, deps: [MAT_DATE_LOCALE]},
+    {provide: MAT_DATE_FORMATS, useValue: MAT_MOMENT_DATE_FORMATS},
+  ],
+})
+export class TeacherCardComponent implements OnInit {
+  @Input() teacher: Teacher;
+
+  private fileToUpload: string | ArrayBuffer;
+  private avatarImg: string | ArrayBuffer;
+  private maxAge = this.teachServise.checkAgeDate();
+  private editTeacher: AbstractControl;
+  private subscriptAvatar: any;
+
+  constructor(private teachServise: TeachersService,
+              private validServ: ValidationService,
+              private formBuilder: FormBuilder,
+              private notify: NotificationService) {
+                this.editTeacher = this.formBuilder.group({
+                   firstname: ['', ],
```

Comment:
can we replace  `''` with default values? So we can create one method e.g. 
```typescript
createFormWitDefaultValues() {
    this.editTeacher = this.formBuilder.group({
                   firstname: [this.teacher.firstname, ],
                   ......
}
```

Reply:
and call it oninit

Reply:
maybe now it will be work
earlier it was impossible